src/lib/containers/download_list/DownloadListTable.tsx

```diff
@@ -0,0 +1,264 @@
+import React, { useState, useEffect } from 'react'
+import { EntityHeader } from '../../utils/jsonResponses/EntityHeader'
+import {
+  getDownloadList,
+  getEntityHeader,
+  getFiles,
+  deleteDownloadListFiles,
+  deleteDownloadList,
+} from '../../utils/SynapseClient'
+import { library } from '@fortawesome/fontawesome-svg-core'
+import { faTrash } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import HasAccess from '../HasAccess'
+import { calculateFriendlyFileSize } from '../../utils/functions/calculateFriendlyFileSize'
+import DownloadDetails from './DownloadDetails'
+import { Reference } from '../../utils/jsonResponses/ReferenceList'
+import useGetProfiles from '../../utils/hooks/useGetProfiles'
+import { PaginatedResults } from '../../utils/jsonResponses/PaginatedResults'
+import { BatchFileRequest } from '../../utils/jsonResponses/BatchFileRequest'
+import { BatchFileResult } from '../../utils/jsonResponses/BatchFileResult'
+import moment from 'moment'
+import * as ReactBootstrap from 'react-bootstrap'
+import { DownloadList } from '../../utils/jsonResponses/Download/DownloadList'
+import UserCard from '../UserCard'
+import {
+  FileHandleAssociation,
+  FileHandleAssociateType,
+} from '../..//utils/jsonResponses/FileHandleAssociation'
+import './DownloadList.scss'
+import { CreatePackage } from './CreatePackage'
+
+library.add(faTrash)
+
+type DownloadListTableData = {
+  references?: PaginatedResults<EntityHeader>
+  batchFileResult?: BatchFileResult
+  downloadList?: DownloadList
+}
+
+type LoadingState = boolean
+export type DownloadListTableProps = {
+  token?: string
+}
+
+export const TESTING_TRASH_BTN_CLASS = 'TESTING_TRASH_BTN_CLASS'
+export const TESTING_CLEAR_BTN_CLASS = 'TESTING_CLEAR_BTN_CLASS'
+
+export default function DownloadListTable(props: DownloadListTableProps) {
+  // https://reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables
+  let [data, setData] = useState<DownloadListTableData>({
+    references: undefined,
+    batchFileResult: undefined,
+    downloadList: undefined,
+  })
+  // https://reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables
+  let [isLoading, setIsLoading] = useState<LoadingState>(true)
+  const { token } = props
+  const { references, batchFileResult, downloadList } = data
+  const requestedFiles =
+    (batchFileResult && batchFileResult.requestedFiles) || []
+  const ownerIds: any = requestedFiles
+    .map(el => el.fileHandle && el.fileHandle.createdBy)
+    .filter(el => el)
```

Comment:
does this filter out falsey values?

Reply:
Yes, but I realize now thats a bit clunky, I changed it to instead filter out the items that have createdBy set and then map them out to just that value